smoke_test/tests/sam_smoke_test_case.py

```diff
@@ -0,0 +1,23 @@
+from functools import cache
+from unittest import TestCase
+from urllib.parse import urljoin
+
+import requests
+from requests import Response
+
+
+class SamSmokeTestCase(TestCase):
+    SAM_HOST = None
+    USER_TOKEN = None
+
+    @staticmethod
+    def build_sam_url(path: str) -> str:
+        assert SamSmokeTestCase.SAM_HOST, "ERROR - SamSmokeTests.SAM_HOST not properly set"
+        return urljoin(f"https://{SamSmokeTestCase.SAM_HOST}", path)
+
+    @staticmethod
+    @cache
+    def call_sam(url: str, user_token: str = None) -> Response:
+        """Function is memoized so that we only make the call once"""
```

Comment:
interesting! so this means that we can have multiple test cases for a single api call?

Reply:
indeed, saves making repetitive calls to reduce load on Sam and improve performance.